### ISSUE TITLE
BVN

### ISSUE THREAD

**Issue by adebeun** [retrospect-addon/plugin.video.retrospect#1507]
### What Channel would you like to add and why?

BVN, BVN staat voor ‘het Beste van Vlaanderen en Nederland’ en is daarmee de Nederlandstalige publieke tv-zender in het buitenland.
BVN is currently available via satellite, but from 1 July 2021 this satellite broadcast to most of the world will be discontinued. Streaming is then the only option. BVN make an Android app, but not one that functions on an Android Smart TV.

### Provide some details for the channel
- Name of the channel in English: BVN
- Name of the channel in native language:  BVN
- Website for the channel: https://www.bvn.tv/
- If available link to the Android app for the VOD content of channel: https://play.google.com/store/apps/details?id=tv.bvn.app
- Link to the logo of the channel
 https://www.bvn.tv/wp-content/themes/bvn/assets/img/bvn-logo.png

### Check which features should be implemented
- [X] Video on-demand content
- [X] TV live streams
- [ ] Use your account for the VOD services from Retrospect
- [ ] Access the premium content if you are eligible  

### Additional information

It would be nice to get the live stream to work first. Video on-demand could be a lower priority and perhaps added at a later time.

**Comment by basrieter** [comment]
Als ik het goed lees is het binnenkort enkel nog maar NPO: https://www.bvn.tv/wijzigingen/

Heeft het dan nog wel toegevoegde waarde? 

Het is trouwens een duplicate van #1203

**Comment by adebeun** [comment]
My apologies for not finding the earlier request, I only searched the open issues.
I don't know whether BVN offers additional value over NPO. I expect it will become a kind of "Best of NPO" compilation. I guess time will tell. As the earlier issue notes, BVN is quite popular with expats, that is certainly true with the seniors here in New Zealand. I'm trying to find and easy to use alternative to the satellite box that will soon be obsolete.

I had a look at how these plug-ins are implemented, with the intention of adding BVN myself. But this looks much more involved than I anticipated and will require a steep learning curve on my part.
How much effort is involved in implementing (and maintaining) a new channel? I can potentially help with that.

**Comment by adebeun** [comment]
I tried to play NPO in New Zealand, but NPO is geo-blocked and actually shows a message referring to BVN instead:

_Vanwege uitzendrechten is het niet mogelijk om deze uitzending buiten Nederland te bekijken.
We verwijzen je graag door naar de livestream van BVN. BVN toont een uitgebreide selectie van de tv-programma's van NPO.
Start hier de BVN livestream._

So this is perhaps a sufficient reason to support BVN as well.

**Comment by basrieter** [comment]
No worries. Let's see how it goes. I don't feel like adding the VRT stuff if they are removing it.  So perhaps only the NPO stuff.

**Comment by bruggieinoz** [comment]
I would like to support this request for BVN. It is certainly not being discontinued. What the article refers to, is that satellite coverage will/has been partially removed, thus the desire for a streaming service alternative. Yes, BVN is a "best of" NPO, but compiled for all the expats and would have something to offer in our time zone. Like adebeun this is a very steep learning curve for me, but happy to help where I can. I primarily want to support the older Dutch community in Australia, of which many have been watching BVN for years. The smartphone app is just not a feasible replacement for this generation. Your help would be much appreciated

**Comment by ErikBrown2** [comment]
After adding a separate request for BVN just now, I found this existing request. I live in the Middle East and for Dutch people here who like to watch NPO, programs are regularly geo blocked. But often the same programs that are geo blocked on NPO are actually available for viewing on BVN. But BVN is only available for us for viewing on a PC since the BVN website does not support the Chromecast. So if Retrospect would support BVN, then it will be a really great way of viewing these programs that we cannot see on NPO due to the geographic restrictions.

**Comment by basrieter** [comment]
I will see what is needed. Almost summer break, so I have some time.

**Comment by bruggieinoz** [comment]
I just wanted to inform you that I actually can now view the BVN Livestream with the "Catch-up-TV-and others" Add-on. I remember seeing this commented earlier, but a couple of days ago I could not get it to function. It works now.
@ErikBrown2 I have this installed on a Raspberry PI 4. This is all self-starting and after bootup it outputs the BVN video signal. All in a similar fashion as the old satellite set-top-box. @basrieter thank you for considering adding this to Retrospect. There may still be benefits doing that, but my BVN "mission has been accomplished". Happy to share my setup details with anyone interested via PM.

**Comment by ErikBrown2** [comment]
@bruggieinoz Where did you find that "Catch-up-TV-and others" Add-on? I searched for it but only found a website that Avast Anti Virus told me that it had blocked. Is it a trustworthy add-on?

**Comment by bruggieinoz** [comment]
@ErikBrown2. Sorry, I did not realise Github has no (longer) PM. I followed this excellent guide. https://mediaexperience.com/raspberry-pi-xbmc-with-raspbmc/ <https://mediaexperience.com/raspberry-pi-xbmc-with-raspbmc/>
This enabled me to have a BVN Live Video Signal from a Raspberry Pi 4 within the hour (not counting the many hours testing this out on Kodi Windows prior to that).
Catch-up, like Retrospect and all other required add-ons come from the Kodi repository. Apologies for highjacking this thread, as this probably does not belong here

**Comment by basrieter** [comment]
Please install this zip in Kodi Matrix (not Leia!): https://github.com/retrospect-addon/plugin.video.retrospect/archive/refs/heads/channel-1507.zip

And let me know if the TV Shows work? I can't really test possible geo blocking, as I live in The Netherlands.

**Comment by bruggieinoz** [comment]
Thank you basrieter. I installed the zip, but I can only see a selection of recorded BVN streams, not the BVN Livestream? I can play some of the recorded streams. Some others I cannot due to geo blocking

**Comment by basrieter** [comment]
The live stream is not in there yet. For now I need to know:

1. Do all episodes that play on the site work for you in Retrospect
2. Do those that can't be played on the site due to geo restrictions, also fail in Retrospect (please provide examples of such shows).

**Comment by bruggieinoz** [comment]
here’s some results.

**Under “BVN”**
Under a lot of the selections, I get "No episodes found for the selected item”

for the ones I can select:

2Doc Kort: Samenzwevers?: OK
Nu te zien! -Surinaamse school: OK
Nederland op film - eindelijk weekend: OK
Het video dagboek van Anne Frank: Dit programma mag niet bekeken worden vanaf jouw locatie (33). Same error when using BVN website

**Under NOS.nl**
Politie Amsterdam: deel beelden met ons, niet op sociale media OK

**NPO Start/Live TV**
NPO1 NOS Journaal: unspecified error retrieving streams

Would this be of help?
https://github.com/Catch-up-TV-and-More/plugin.video.catchuptvandmore/blob/dev/resources/lib/channels/wo/bvn.py
I can watch BVN Live with this addon

**Comment by ErikBrown2** [comment]
Some programs are fine. However, it appears that BVN is also working on implementing geo restrictions. Some programs have it already implemented. A sample is the program https://www.bvn.tv/programma/denkend-aan-holland/

The episode "'t Groene Hart" plays fine both on the website and Retrospect. The other two episodes result on the website as well as in Retrospect in "Dit programma mag niet bekeken worden vanaf jouw locatie (33).". On Retrospect also the message "Retrospect - Error The selected item does not have any streams associated." is shown. 

There is an issue with some other programs. When I tried them initially, they did not play at all, neither on Retrospect nor on the website. But when I tried 5 minutes later, they do play on the website but not on Retrospect. When it does not play on the website, then there is no error message shown at all, the program just does not play. On retrospect the error message is "No epidsodes found for the selected item". Samples of such programs are 

"De Gert & Hermien story", 
https://www.bvn.tv/programma/de-gert-hermien-story/
https://www.bvn.tv/programma/de-gert-hermien-story/VPWON_1332942/

2Doc: Dood in de Bijlmer
https://www.bvn.tv/programma/2doc-dood-in-de-bijlmer/
https://www.bvn.tv/programma/2doc-dood-in-de-bijlmer/VPWON_1322563/

40 jaar TV show
https://www.bvn.tv/programma/40-jaar-tv-show/
https://www.bvn.tv/programma/40-jaar-tv-show/AT_300000257/

I do not think that it is a cache issue on Retrospect. I restarted Kodi and still the programs are not available in Retrospect. They keep on playing on the website. If you want more sample programs, then let me know.

**Comment by ErikBrown2** [comment]
I have uploaded a Retrospect debug log as https://paste.kodi.tv/irawenorof

For the log I tried to play "40 jaar TV show". When it should have shown the episode listing, it showed the message "No episodes found for the selected item". The program played fine on the website.

**Comment by basrieter** [comment]
Ok, thanks all for the feedback. There was a minor issue with some shows that resulted in "no episodes", but that should be fixed. Please redownload, install and try again: https://github.com/retrospect-addon/plugin.video.retrospect/archive/refs/heads/channel-1507.zip

> here’s some results.
> **Under NOS.nl**
> Politie Amsterdam: deel beelden met ons, niet op sociale media OK
> 
> **NPO Start/Live TV**
> NPO1 NOS Journaal: unspecified error retrieving streams

@bruggieinoz These two are not in BVN right?

**Comment by basrieter** [comment]
Also add the Live stream now (at the bottom of the list).

**Comment by ErikBrown2** [comment]
Het werkt prima! Alle programma's werken en live TV ook. Heel mooi.

**Comment by bruggieinoz** [comment]
@basrieter yes, the NOS and NPO examples where not part of BVN. Apologies for the confusion. BVN Live werkt inderdaad prima in Oz. Heel erg bedankt. Nice job

**Comment by basrieter** [comment]
Ok, so let me finish up the artwork and start with publishing that. I now have these:

Logo:
![Logo](https://raw.githubusercontent.com/retrospect-addon/plugin.video.retrospect/channel-1507/channels/channel.nos/bvntv/bvntvicon.png)

Fanart:
![Fanart](https://raw.githubusercontent.com/retrospect-addon/plugin.video.retrospect/channel-1507/channels/channel.nos/bvntv/bvntvfanart.jpg)

Post: 
![image](https://user-images.githubusercontent.com/1310443/125193638-c3301700-e24d-11eb-85f8-eae8c7b54cb9.png)

**Comment by basrieter** [comment]
it will be in the next release somewhere this week.

**Comment by ErikBrown2** [comment]
Nice. Thank you very much for adding the channel.
